### PR TITLE
Fix stream sockets

### PR DESCRIPTION
**Pull request by c-cube**
the code would compile but just not work at all (probably it would pick some garbage value for the socket as there's and out of bound error.)

### CODE REVIEW

File: zmq/src/caml_zmq_stubs.c [hunk 1]
```diff
@@ -138,15 +138,19 @@ static int const socket_type_for_kind[] =  {
...
 };
 
 CAMLprim value caml_zmq_socket(value ctx, value socket_kind) {
     CAMLparam2 (ctx, socket_kind);
     CAMLlocal1 (sock_value);
     void *socket;
 
-    socket = zmq_socket(CAML_ZMQ_Context_val(ctx), socket_type_for_kind[Int_val(socket_kind)]);
+    int kind = Int_val(socket_kind);
+    assert (kind <= ZMQ_STREAM);
```
Comment (Leonidas-from-XIV) [inline zmq/src/caml_zmq_stubs.c:151]: Maybe also add `kind >= ZMQ_PAIR`?

**Review by Leonidas-from-XIV — COMMENTED** [review]
Argh, this is an annoying piece of code, split between OCaml and C with no proper tooling-enforced link. The code itself looks good, could you add an entry to the changelog?

**Comment by c-cube** [comment]
done! good point.

It's funny because when I started suspecting the bug, I looked at the implementation, and the OCaml integer _had the correct value_ (11). It's just another place where it was lacking :sweat_smile:

**Review by Leonidas-from-XIV — APPROVED** [review]
Looks good and having an example is certainly appreciated!

**Comment by c-cube** [comment]
I forgot to mention: you can just communicate with the example with `netcat` :)

**Pull request merged**